### PR TITLE
update output file type guidelines

### PR DESCRIPTION
**Pull request by nvnieuwk**

### CODE REVIEW

**Comment by nvnieuwk** [comment]
The preview doesn't seem to show the changes... any ideas why @mashehu ?

File: sites/docs/src/content/docs/guidelines/components/modules.md [hunk 1]
```diff
@@ -464,7 +464,18 @@ tuple val(meta), path('*.tab'), emit: tab,  optional: true
 
 ### One output channel per output file type
 
-Each output file SHOULD be emitted in its own channel (and no more than one), along with the `meta` map if provided ( the exception is the versions.yml ).
+Each output file type SHOULD be emitted in its own channel (and no more than one), along with the `meta` map if provided ( the exception is the versions.yml ).
+
+:::info{title="Rationale" collapse}
+The instruction to emit each output file type in its own channel ensures better organization and manageability of the output files. By assigning a dedicated channel to each file type, you can prevent clutter and potential confusion that might arise from mixing different file types in a single channel. This approach also simplifies the process of retrieving and processing specific types of output, as each type can be easily identified and accessed within its designated channel.
+
+In some cases the file format can be different between files of the same type (e.g. indices: `.bai` and `.crai`). These different file formats SHOULD be part of the same output channel since they are the same file type.
```
Comment (mashehu) [inline sites/docs/src/content/docs/guidelines/components/modules.md:472]: Imo this should be outside of the rationale block, because it contains guidelines

File: sites/docs/src/content/docs/guidelines/components/modules.md [hunk 2]
```diff
@@ -464,7 +464,18 @@ tuple val(meta), path('*.tab'), emit: tab,  optional: true
 
 ### One output channel per output file type
 
-Each output file SHOULD be emitted in its own channel (and no more than one), along with the `meta` map if provided ( the exception is the versions.yml ).
+Each output file type SHOULD be emitted in its own channel (and no more than one), along with the `meta` map if provided ( the exception is the versions.yml ).
+
+:::info{title="Rationale" collapse}
+The instruction to emit each output file type in its own channel ensures better organization and manageability of the output files. By assigning a dedicated channel to each file type, you can prevent clutter and potential confusion that might arise from mixing different file types in a single channel. This approach also simplifies the process of retrieving and processing specific types of output, as each type can be easily identified and accessed within its designated channel.
```
Comment (mashehu) [inline sites/docs/src/content/docs/guidelines/components/modules.md:470]: too wordy, imo

Comment (nvnieuwk) [reply sites/docs/src/content/docs/guidelines/components/modules.md]: Is this better?

File: sites/docs/src/content/docs/guidelines/components/modules.md [hunk 3]
```diff
@@ -466,15 +466,15 @@ tuple val(meta), path('*.tab'), emit: tab,  optional: true
...
-The instruction to emit each output file type in its own channel ensures better organization and manageability of the output files. By assigning a dedicated channel to each file type, you can prevent clutter and potential confusion that might arise from mixing different file types in a single channel. This approach also simplifies the process of retrieving and processing specific types of output, as each type can be easily identified and accessed within its designated channel.
-
 In some cases the file format can be different between files of the same type (e.g. indices: `.bai` and `.crai`). These different file formats SHOULD be part of the same output channel since they are the same file type.
 
 ```groovy
 tuple val(meta), path("*.{bai,crai}"), emit: index
 ```
 
+:::info{title="Rationale" collapse}
+The instruction to emit each output file type in its own channel ensures better organization and manageability of the output files. This approach also simplifies the process of retrieving and processing specific types of output, as each type can be easily identified and accessed within its designated channel.
```
Comment (mashehu) [inline sites/docs/src/content/docs/guidelines/components/modules.md:476]:
```suggestion
This approach also simplifies the process of retrieving and processing specific types of output, as each type can be easily identified and accessed within its designated channel.
```
maybe add an example in the rationale how to access it.

**Review by mashehu — APPROVED** [review]

File: sites/docs/src/content/docs/guidelines/components/modules.md [hunk 4]
```diff
@@ -464,7 +464,30 @@ tuple val(meta), path('*.tab'), emit: tab,  optional: true
 
 ### One output channel per output file type
 
-Each output file SHOULD be emitted in its own channel (and no more than one), along with the `meta` map if provided ( the exception is the versions.yml ).
+Each output file type SHOULD be emitted in its own channel (and no more than one), along with the `meta` map if provided ( the exception is the versions.yml ).
+
+In some cases the file format can be different between files of the same type (e.g. indices: `.bai` and `.crai`). These different file formats SHOULD be part of the same output channel since they are the same file type.
```
Comment (jfy133) [inline sites/docs/src/content/docs/guidelines/components/modules.md:469]: You should sort of specify that they can go in the same channel, as they 1) serve the same purpose and 2) the two formats are mutually exclusive.

File: sites/docs/src/content/docs/guidelines/components/modules.md [hunk 5]
```diff
@@ -464,7 +464,30 @@ tuple val(meta), path('*.tab'), emit: tab,  optional: true
...
+
+```groovy
+tuple val(meta), path("*.txt"), emit: txt
+```
+
+The output files can be accessed like this:
+
+```groovy
+TXT_MODULE.out.txt
+```
```
Comment (jfy133) [inline sites/docs/src/content/docs/guidelines/components/modules.md:488]: I don't really get this example... usn't this just describing output channels?

**Comment by mashehu** [comment]
Yes, it's in the wrong subsite. We currently don't have a preview for that for now. You can have a look at `npm run dev -w sites/docs`

File: sites/docs/src/content/docs/guidelines/components/modules.md (hunk 5 above)
Comment (nvnieuwk) [reply sites/docs/src/content/docs/guidelines/components/modules.md:488]: Yes I'm not sure what would be a better example here. Do you have any ideas?

File: sites/docs/src/content/docs/guidelines/components/modules.md (hunk 4 above)
Comment (nvnieuwk) [reply sites/docs/src/content/docs/guidelines/components/modules.md:469]: Done!

**Comment by nvnieuwk** [comment]
@jfy133 is this okay for you?

**Comment by jfy133** [comment]
Sorry for the delay @nvnieuwk ! Chaos everywhere

**Comment by jfy133** [comment]
To be clear @nvnieuwk I vbetter description of what I meant above in the comment thread

File: sites/docs/src/content/docs/guidelines/components/modules.md (hunk 5 above)
Comment (nvnieuwk) [reply sites/docs/src/content/docs/guidelines/components/modules.md:488]: Good idea! On it!

Comment (nvnieuwk) [reply sites/docs/src/content/docs/guidelines/components/modules.md]: @jfy133 I updated the example, is this better?

File: sites/docs/src/content/docs/guidelines/components/modules.md [hunk 6]
```diff
@@ -464,7 +464,30 @@ tuple val(meta), path('*.tab'), emit: tab,  optional: true
 
 ### One output channel per output file type
 
-Each output file SHOULD be emitted in its own channel (and no more than one), along with the `meta` map if provided ( the exception is the versions.yml ).
+Each output file type SHOULD be emitted in its own channel (and no more than one), along with the `meta` map if provided ( the exception is the versions.yml ).
+
+In some cases the file format can be different between files of the same type (e.g. indices: `.bai` and `.crai`). These different file formats SHOULD be part of the same output channel since they are they serve the same purpose and are mutually exclusive.
```
Comment (jfy133) [inline sites/docs/src/content/docs/guidelines/components/modules.md:469]:
```suggestion
In some cases the file format can be different between files of the same type or for the same function (e.g. indices: `.bai` and `.crai`). These different file formats SHOULD be part of the same output channel since they are they serve the same purpose and are mutually exclusive.
```

File: sites/docs/src/content/docs/guidelines/components/modules.md [hunk 7]
```diff
@@ -464,7 +464,30 @@ tuple val(meta), path('*.tab'), emit: tab,  optional: true
...
+tuple val(meta), path("*.{bai,crai}"), emit: index
+```
+
+The output files can be accessed like this:
+
+```groovy
+SAMTOOLS_INDEX.out.index
+```
+
+:::
```
Comment (jfy133) [inline sites/docs/src/content/docs/guidelines/components/modules.md:490]:
```suggestion

Regardless whether they are a `bai` or `crai` as downstream SAMTOOLS modules should accept either without an issue. 

:::
```

**Review by jfy133 — APPROVED** [review]
